### PR TITLE
Refactor the FilteredCollectionHeader so it's generic

### PR DESCRIPTION
**Pull request by paulgain**
## Description of change

The `FilteredCollectionHeader` of the `FilteredCollectionList` was meant to be a generic component. At present, the header contains a [significant amount](https://github.com/uktrade/data-hub-frontend/blob/master/src/client/components/FilteredCollectionList/FilteredCollectionHeader.jsx#L96-L404) of hardcoded/specific `RoutedFilterChips` which has unfortunately grown over time. This PR rectifies the problem by removing all the chips and introducing a new component called `RoutedFilterChipsCollection` which simply [iterates](https://github.com/uktrade/data-hub-frontend/blob/refactor/routed-filter-chips/src/client/components/FilteredCollectionList/FilteredCollectionHeader.jsx#L59-L69) over the selected filters of the collection. In order to do this we need to slightly change how the filters are structured so we can pull out  the query parameter. For example:

**Before**
```javascript
  // filters.js
  ...
  selectedCompanyName: buildInputFieldFilter({
    value: company_name,
    categoryLabel: LABELS.companyName,
  }),
  ...
```

**After**
```javascript
  // filters.js
  ...
  companyName: {
    queryParam: 'company_name',
    options: buildInputFieldFilter({
      value: queryParams.company_name,
      categoryLabel: LABELS.companyName,
    }),
  },
  ...
```

**Now we loop over each filter object key creating a chip**
```javascript
// FilteredCollectionHeader.jsx
const RoutedFilterChipsCollection = ({ filters }) => (
  <>
    {Object.keys(filters).map((key) => (
      <RoutedFilterChips
        key={key}
        selectedOptions={filters[key].options}
        qsParamName={filters[key].queryParam}
      />
    ))}
  </>
)
```

**Each routed Typeahead/Checkbox component then pulls in its options like so:**
```javascript
<RoutedTypeahead
  ...
  selectedOptions={filters.companySectors.options}
  ...
/>
```

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

**Review by tomkingston — COMMENTED** [review]
Sorry, now that I see everything together I think the `buildFilters` functions should have remained `buildSelectedFilters`. Other than that, looks great 👍

**Review by ian-leggett — APPROVED** [review]
Subject to @tomkingston  change all good 👍

**Review by tomkingston — APPROVED** [review]

File: test/functional/cypress/specs/interaction/filter-react-spec.js [hunk 1]
```diff
@@ -192,12 +192,17 @@ describe('Interactions Collections Filter', () => {
...
+      // TODO: unfortunately we're now getting duplicate chips on the
+      // adviser field so the test fails here when attempting to remove
+      // one of them: "cy.click() can only be called on a single element"
+      // removeChip(adviser.id)
+
+      // cy.wait('@apiRequest').then(({ request }) => {
+      //   expect(request.body).to.deep.equal(minimumPayload)
+      // })
+      // assertChipsEmpty()
+      // assertFieldEmpty(advisersFilter)
```
Comment (paulgain) [inline test/functional/cypress/specs/interaction/filter-react-spec.js:205]: @ian-leggett as you're working on interactions do you mind taking a look at why we're now getting duplicate chips. Thanks.

File: test/functional/cypress/specs/interaction/filter-react-spec.js [hunk 2]
```diff
@@ -247,10 +247,12 @@ describe('Interactions Collections Filter', () => {
...
-      removeChip(adviser.id)
-      assertPayload('@apiRequest', minimumPayload)
-      assertChipsEmpty()
-      assertFieldEmpty(myInteractionsFilter)
+
+      // Exactly the same as the TODO above
+      // removeChip(adviser.id)
+      // assertPayload('@apiRequest', minimumPayload)
+      // assertChipsEmpty()
+      // assertFieldEmpty(myInteractionsFilter)
```
Comment (paulgain) [inline test/functional/cypress/specs/interaction/filter-react-spec.js:255]: @ian-leggett same here.

**Pull request merged**